tests/bdd/features/snapshot/create/feature.feature

```diff
@@ -0,0 +1,78 @@
+Feature: Volume Snapshot Creation
+
+  Background:
+    Given a deployer cluster
+
+  Scenario Outline: Snapshot creation of a single replica volume
+    Given we have a single replica <publish_status> volume where the replica is <replica_location> to the target
+    When we try to create a snapshot for the volume
+    Then the snapshot creation should be successful
+    And the snapshot status ready to clone should be true
+    And the snapshot source size should be equal to the volume size
+    And the snapshot size should be equal to the volume size
+    And the snapshot timestamp should be in the RFC 3339 format
+    And the snapshot timestamp should be within 1 minute of creation (UTC)
+    And the volume snapshot state has a single online replica snapshot
+    And the replica snapshot's timestamp should be in the RFC 3339 format
+    And the replica snapshot's timestamp should be within 1 minute of creation (UTC)
+    Examples:
+      | publish_status | replica_location |
+      | published      | remote           |
+      | published      | local            |
+      | unpublished    | remote           |
+      | unpublished    | local            |
+
+  Scenario: Snapshot creation for a single replica volume fails when the replica is offline
+    Given we have a single replica volume
+    And the replica is offline
+    And volume is in faulted state
+    When we try to create a snapshot for the volume
+    Then the snapshot creation should fail with preconditions failed
+    And no snapshot should be listed
+
+  Scenario: Snapshot creation for multi-replica volume is not supported
+    Given we have a multi-replica volume
+    When we try to create a snapshot for the volume
+    Then the snapshot creation should be fail with preconditions failed
+
+  Scenario: Existence of snapshot after node restart
+    Given we have a single replica volume
+    And a successfully created snapshot
+    When we stop the node hosting the snapshot
+    Then the replica snapshot should be reported as offline
+    When we resume the node hosting the snapshot
+    Then the replica snapshot should be reported as online
+
+  Scenario: Retrying snapshot creation until it succeeds
+    Given we have a single replica volume
+    And the volume is published on node A
+    And the node A is brought down
+    When we try to create a snapshot for the volume
+    Then the snapshot creation should fail
+    When the node A is brought back
+    And we retry the same snapshot creation
+    Then the snapshot creation should be successful
+
+  Scenario: Snapshot creation of a single replica volume after its source is deleted
+    Given we have a single replica volume
+    When a snapshot is created for the volume
+    When the volume is deleted
+    And we retry the same snapshot creation
+    Then it should fail with already exists
```

Comment:
This one looks confusing to me. Are we testing that `same snapshot/id creation should fail with eexist` i.e. the last scenario down in this file, or we are testing that `any snapshot creation after volume deletion should fail` ?

Reply:
we're testing that it fails with eexists even when the volume has been deleted